x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/CachedBlob.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.blobstore.cache;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.Map;
+
+public class CachedBlob implements ToXContent {
+
+    /**
+     * Sentinel {@link CachedBlob} indicating that searching the cache index returned an error.
+     */
+    public static final CachedBlob CACHE_NOT_READY = new CachedBlob(null, null, null, "CACHE_NOT_READY", null, BytesArray.EMPTY, 0L, 0L);
+
+    /**
+     * Sentinel {@link CachedBlob} indicating that the cache index definitely did not contain the requested data.
+     */
+    public static final CachedBlob CACHE_MISS = new CachedBlob(null, null, null, "CACHE_MISS", null, BytesArray.EMPTY, 0L, 0L);
+
+    private static final String TYPE = "blob";
+
+    private final Instant creationTime;
+    private final Version version;
+    private final String repository;
+    private final String name;
+    private final String path;
+
+    private final BytesReference bytes;
+    private final long from;
+    private final long to;
+
+    public CachedBlob(
+        Instant creationTime,
+        Version version,
```

Comment:
The actual format of blobs in the index is not settled but is relatively easy to change. I can see value in recording the version of Elasticsearch that filled the cache since this should enable BWC should we need to change the format in future.

Reply:
> this should enable BWC should we need to change the format in future.

Yes, that was the main reason I added this information. The version of the index is already stored in the `index.version.created` metadata which is set back during restore and never modified.